### ISSUE TITLE
use of `puppet agent --enable` and `puppet agent --disable`

### ISSUE THREAD

**Issue by neoice** [mozilla-it/puppetctl#9]
I'm investigating mcollective and it is aware of the Puppet3 `--enable` and `--disable` flags.

```
root@test3:~# puppet agent --disable "this is a test message"
root@test3:~# cat /var/lib/puppet/state/agent_disabled.lock  
{"disabled_message":"this is a test message"}#
```

using these flags allows mcollective to perform puppet runs on enabled hosts but not disabled hosts. in my current infrastructure using puppetctl, mcollective will run on puppetctl disabled hosts.

`puppet agent --disable` isn't as smart as `puppetctl`: there's no timed disables or motd modification.

if I wind up modifying puppetctl to support mcollective, are you interested in including the changeset in upstream? I think it might be as simple as _also_ doing a `puppet agent --{en,dis}able` for the corresponding puppetctl actions.

**Comment by neoice** [comment]
doing a scrub of my open Github issues...

I've been running [this](https://github.com/neoice/puppetctl/commit/8f8f40636c1b90866eb2755ea670ef30deba5c41) in production for over a year now. I don't think I've had any problems.

are there any updates from Mozilla that you can publish? no new updates since Sep 2013!

**Comment by bhourigan** [comment]
Yeah of course. I'll push an update now. Although, we're still not using puppet's internal enable and disable toggle as we run puppet from cron.

**Comment by neoice** [comment]
I've been building a lot of automation with mcollective, but so far, I don't think I actually need the `puppet` mco agent at all. massive Puppet runs cause all kinds of bottlenecks (VM disk IO, HTTP request timeouts on Puppetmaster, HTTP requests/disk IO on rpm repos, etc).

I could probably remove the `--enable`/`--disable` stuff with no negative impacts. in fact, it annoys a few of my team mates because `puppet agent --test --noop` will refuse to run in disabled state!

**Comment by bhourigan** [comment]
I just merged an update for you

**Comment by neoice** [comment]
awesome. I currently have a *lot* of disabled hosts, so I might not push it out for a few weeks.

are you able to explain why the path was changed? just good housekeeping?

**Comment by bhourigan** [comment]
There is a comment inside puppetctl explaining the reason for the change. On RHEL /var/run is cleared at boot, so that had the unintentional side effect of re-enabling puppet. It wasn't a common use case, but it was a surprise for sure.
